### ISSUE TITLE
gtkui: add saving and loading of UI layouts

### ISSUE THREAD

**Issue by cboxdoerfer** [Alexey-Yakovenko/deadbeef#1362]
I already implemented most of that feature but there are some details I'd like to get your opinion about.

My current implementation adds two new actions: Load Layout and Save Layout which can be triggered via the menubar or hotkeys. Both spawn a gtk filechooser where the user can decide where to save/load the layout to/from. A layout file looks like that: ```<magic_number><api_major><api_minor><length_of_layout_string><layout_string>```

Here are the questions:
### 1) What file extension do you guys prefer?
Here are two suggestions: ```*.dblayout, *.dbth ([th]eme)```

### 2) Should we make them UI agnostic?
With the cocoa UI or android UI in mind it could make some sense to make those layouts not only work with GTKUI. But I don't know if there are plans for that in other UIs and if the widgets of other toolkits would even work with the current layout format. If not we might need to use a more specific file extension name (e.g. *.dbgl ([d]ead[b]eef[g]tk[l]ayout) to avoid confusion in the future.

### 3) Or do you prefer a different solution than saving/loading layouts to/from unique files?
I also thought about saving those layouts in the deadbeef config file, but personally I like the file approach much better. This way users can more easily share their layouts.

**Comment by Alexey-Yakovenko** [comment]
>  A layout file looks like that: <magic_number><api_major><api_minor><length_of_layout_string><layout_string>

You mean binary? Bad idea IMO. Why not just put them in JSON, and have the version number / magic id as normal fields?

1) .ddb instead of db would be more practical, to reduce the confusion versus "database".

2) No -- cocoa and android UI don't support layout customization (yet), and when/if they do -- it's not guaranteed they would be using the same structure.

3) One layout per file is definitely the right way to go.

**Comment by cboxdoerfer** [comment]
Yes, right now they are binary, because I didn't like the idea of users to mess with them direcly and it was also easier/quicker to implement. But of course I can use JSON instead.

1) Ok, then something like that: *.ddblayout?

**Comment by Alexey-Yakovenko** [comment]
> because I didn't like the idea of users to mess with them direcly and it was also easier/quicker to implement. 

That would be a very wrong thing to do. Let users do whatever they wish. It's their files, etc.
And using the easily parseable standard format like JSON is the right thing.

> 1) Ok, then something like that: *.ddblayout?

looks ok to me.
